worker/upgradesteps/worker.go

```diff
@@ -191,7 +191,7 @@ func (w *upgradesteps) run() error {
 		if w.pool, err = w.openState(); err != nil {
 			return err
 		}
-		defer w.pool.Close()
+		defer func() { _ = w.pool.Close() }()
```

Comment:
should we have a pattern of "log an error if not nil, but otherwise ignore it ?"

Reply:
Yep.